src/types/tuple.rs

```diff
@@ -276,20 +610,17 @@ impl<'a> DoubleEndedIterator for PyTupleIterator<'a> {
     }
 }
 
-impl<'a> ExactSizeIterator for PyTupleIterator<'a> {
+impl<'a, 'py> ExactSizeIterator for BorrowedTupleIterator<'a, 'py> {
     fn len(&self) -> usize {
         self.length.saturating_sub(self.index)
     }
 }
 
-impl FusedIterator for PyTupleIterator<'_> {}
-
-impl<'a> IntoIterator for &'a PyTuple {
-    type Item = &'a PyAny;
-    type IntoIter = PyTupleIterator<'a>;
+impl FusedIterator for BorrowedTupleIterator<'_, '_> {}
 
-    fn into_iter(self) -> Self::IntoIter {
-        self.iter()
+impl IntoPy<Py<PyTuple>> for Bound<'_, PyTuple> {
```

Comment:
Do we need/have this for the other native types?

Reply:
This came up as useful because `PyAny::call_method` has `IntoPy<Py<PyTuple>>` as a bound on the `args` parameter.

I think this implementation could be solved generically but it's a little fiddly because `IntoPy<PyObject> for Bound<'_, T>` can easily conflict with `IntoPy<PyObject> for Bound<'_, PyAny>`. So I'd prefer to just special-case this one for now and punt on the more complete formulation (which might be easier to reason about once the gil refs are removed and there's a lower complexity level in these types again).

Reply:
I agree on the reasoning for the specific impl instead of a generic one. But are tracking these missing specific impls somewhere because I think we should add them before release.

Reply:
I'll start an issue tracking bits we might want to polish. 👍

Reply:
For now I just added this as extra bullet to #3684